staging/src/k8s.io/apimachinery/pkg/api/resource/quantity.go

```diff
@@ -592,6 +592,19 @@ func (q *Quantity) Sub(y Quantity) {
 	q.ToDec().d.Dec.Sub(q.d.Dec, y.AsDec())
 }
 
+// Mul multiplies the provided y quantity to the current value. If the current value is zero,
+// the format of the quantity will be updated to the format of y.
+func (q *Quantity) Mul(y Quantity) {
+	q.s = ""
+	if q.IsZero() {
+		q.Format = y.Format
+	}
+	if q.d.Dec == nil && y.d.Dec == nil && q.i.Mul(y.i) {
+		return
+	}
+	q.ToDec().d.Dec.Mul(q.d.Dec, y.AsDec())
```

Comment:
In addition to my other reason for using a scalar -- I believe changing the format will destroy the property that multiplication needs to give the exact result of repeated addition. I don't think you should do this.